bbc/simorgh
# Move AMP iframe and Psammead components to MediaLoader

Overall changes
======
- Moves amp-iframe to MediaLoader for AMP media rendering
- Moves PlayButton, Message, NoScript and Guidance components out of Psammead and into MediaLoader and uplifts them to TS
- Rejigs the way we style landscape and portrait media, moving the bulk of the layout to the `figure` element. This ensures that all content within the `figure` element flows with the aspect-ratio that is set
- Updates snapshots where necessary
- Removes now old `AVPlayer` and `MediaPlayer` components, as well as Psammead components no longer used

Testing
======
1. Visit MediaArticlePage http://localhost:7080/hausa/articles/cdx6ddxw755o.amp?renderer_env=live
2. Confirm the gray BBC placeholder loads before showing a 404/500 window where the media player should be (this is expected as its running locally and attempting to hit a Test endpoint) - You can get around this by appending `?renderer_env=<environment>` to the end of the `ampIframeUrl` variable:
https://github.com/bbc/simorgh/blob/994be3c3b3f54d781954ce6968b8e4a43cb525b3/src/app/components/MediaLoader/index.tsx#L284
3. Disable JS
4. Confirm NoJS messaging appears
5. Repeat for ArticlePage: http://localhost:7080/sport/cricket/articles/cvg37wdd7geo.amp?renderer_env=live 
6. Check Storybook story renders https://amp-iframe-externalembed-constructor--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/components-medialoader--portrait

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
